### PR TITLE
Improve statelessSessions types

### PR DESCRIPTION
**Pull request by timleslie**
The type parameter of this function needs to be more specific to work with `withItemData`.

### CODE REVIEW

**Review by emmatown — CHANGES_REQUESTED** [review]
This isn't right, statelessSessions should not require listKey and itemId

**Comment by timleslie** [comment]
Ah yeah, you're right, I'm going to make a different change which will fix the type errors I'm getting. Will change this PR.

File: packages-next/keystone/src/session/index.ts [hunk 1]
```diff
@@ -115,7 +115,7 @@ export function withItemData<T extends { listKey: string; itemId: string }>(
   };
 }
 
-export function statelessSessions<T>({
+export function statelessSessions<T extends Record<string, any>>({
```
Comment (mitchellhamilton) [inline packages-next/keystone/src/session/index.ts:118]: This restriction seems unnecessary. Iron supports anything that can be json stringified.

File: packages-next/keystone/src/session/index.ts [hunk 2]
```diff
@@ -74,10 +74,10 @@ type FieldSelections = {
   - [ ] We could support additional where input to validate item sessions (e.g an isEnabled boolean)
 */
 
-export function withItemData<T extends { listKey: string; itemId: string }>(
-  createSession: () => SessionStrategy<T>,
+export function withItemData<U extends Record<string, any>>(
+  createSession: () => SessionStrategy<U>,
   fieldSelections: FieldSelections = {}
-): () => SessionStrategy<T & { data: any }> {
+): () => SessionStrategy<U & { listKey: string; itemId: string; data: any }> {
```
Comment (mitchellhamilton) [inline packages-next/keystone/src/session/index.ts:80]: This function being generic seems unnecessary. You could replace U and and the `U & { ... }` with passing in some type which is `{ listKey: string; itemId: string; data: any }`

File: packages-next/keystone/src/session/index.ts (hunk 1 above)
Comment (timleslie) [reply packages-next/keystone/src/session/index.ts:118]: Right, but `withItemData` can't use it unless it's an object.

File: packages-next/keystone/src/session/index.ts (hunk 2 above)
Comment (timleslie) [reply packages-next/keystone/src/session/index.ts:80]: I'm not sure what you mean by this 🤔

File: packages-next/keystone/src/session/index.ts (hunk 1 above)
Comment (mitchellhamilton) [reply packages-next/keystone/src/session/index.ts:118]: Sure withItemData requires an object, why should statelessSessions require it though?

File: packages-next/keystone/src/session/index.ts (hunk 2 above)
Comment (mitchellhamilton) [reply packages-next/keystone/src/session/index.ts:80]: I'm saying remove the `<U extends Record<string, any>>` generic

**Review by emmatown — APPROVED** [review]

**Pull request merged**